### ISSUE TITLE
Keep track of negatives in HData

### ISSUE THREAD

**Issue by tizianocitro** [hypernetwork-research-group/hyperbench#136]
### Is your feature request related to a problem? Please describe.

For some models, it's helpful to have negative and positive hyperedges separated.

### Describe the solution you would like

Add mask for positive and negative samples.
We may need a `remove_negatives()` functions to clean an `hdata` before adding new negatives.
We could also have a `cat_negatives()` that does that automatically.

### Describe alternatives you've considered

Keep a `negative_hyperedge_index` and `positive_hyperedge_index`.

### Additional context

NA.

**Comment by tizianocitro** [comment]
I was reasoning about the following approach.

Given `hdata.hyperedge_index` has shape `[2, num_incidences]`:

```python
hyperedge_index[0]  # node ids per incidence
hyperedge_index[1]  # hyperedge ids per incidence
```

`hdata.y` has shape `[num_hyperedges]`, so labels are per hyperedge, not per incidence. To get positive/negative incidence columns, we can mask through the hyperedge id row:

```python
positive_hyperedge_index = hdata.hyperedge_index[
    :, hdata.y[hdata.hyperedge_index[1]] == 1
]

negative_hyperedge_index = hdata.hyperedge_index[
    :, hdata.y[hdata.hyperedge_index[1]] == 0
]
```

That's enough for readingonly. For `remove_negatives()` or `cat_negatives()` we need a new valid `HData`, which means also slicing `y`, `hyperedge_attr`, `hyperedge_weights`, and rebasing hyperedge ids to contiguous `0..n-1`. The existing `HData.split(...)` already does most of that if we pass selected hyperedge ids. For example, for positives:

```python
positive_ids = torch.where(hdata.y == 1)[0]
positive_hdata = HData.split(hdata, split_hyperedge_ids=positive_ids)
```
